### PR TITLE
Add MTE-2911 MTE-2912 Run Focus tests on iOS 15 and 16

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2911)
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-2912)

## :bulb: Description
Focus smoke tests and full functional tests can be run on iOS 15 and iOS 16 using Github Actions.

Test run: https://github.com/clarmso/metrics-firefox-ios/actions/runs/10714541976/job/29708384512

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: .github/workflows/focus-ios-ui-tests-old.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+          pip3 install virtualenv pipenv
+          gem install xcpretty -v 0.3.0
+          pip install blockkit==1.9.1
+          npm install -g junit-report-merger@7.0.0
+      - name: Setup Xcode
+        id: xcode
+        run: |
+          sudo rm -rf /Applications/Xcode.app
+          sudo rm -fr /Applications/Xcode_16*
+          sudo rm -fr /Applications/Xcode_14*
```
Comment (clarmso) [inline .github/workflows/focus-ios-ui-tests-old.yml:42]: These xcode installations are removed to free up some disk space. The runtime file is large that the Github Actions runner runs out of disk space.

File: .github/workflows/focus-ios-ui-tests-old.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          path: ${{ env.browser }}/junit-focus-ios-*.xml
+          retention-days: 90
+      - name: Upload log file
+        id: upload-log
+        uses: actions/upload-artifact@v4.3.3
+        with:
+          name: ${{ env.browser }}-${{ matrix.ios_simulator }}-xcodebuildlog-${{ github.run_number }}
+          path: ${{ env.browser }}/xcodebuild-*.log
+          retention-days: 90
+      - name: Report to Slack
```
Comment (clarmso) [inline .github/workflows/focus-ios-ui-tests-old.yml:122]: I have not tested this step yet.

File: .github/workflows/focus-ios-ui-tests-old.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          sudo rm -fr /Applications/Xcode_16*
+          sudo rm -fr /Applications/Xcode_14*
+          sudo xcode-select -s /Applications/Xcode_${{ env.xcode_version }}.app/Contents/Developer
+          xcodebuild -version
+          ./checkout.sh
+          ./bootstrap.sh --force
+      - name: Create iOS ${{ matrix.ios_version }} simulator
+        id: setup-simulator
+        run: |
+          sudo xcodes runtimes install "iOS ${{ matrix.ios_version }}"
```
Comment (clarmso) [inline .github/workflows/focus-ios-ui-tests-old.yml:50]: I discovered a more reliable way to install old runtime than before. Fastlane suggests `xcodes`.

**Comment by isabelrios** [comment]
I would change the name for the github action: `.github/workflows/focus-ios-ui-tests-old.yml` looks like it is something old not running.. I would rather add `focus-ios-ui-tests-previous-os-15-16` or something along these lines.. thanks

**Review by isabelrios — APPROVED** [review]
Just a small change in the github action file's title before merging

**Comment by clarmso** [comment]
> I would change the name for the github action: `.github/workflows/focus-ios-ui-tests-old.yml` looks like it is something old not running.. I would rather add `focus-ios-ui-tests-previous-os-15-16` or something along these lines.. thanks

I would say `focus-ios-ui-tests-previous-ios.yml` so that we don't need to update the filename when iOS 15 is no longer supported.

**Comment by isabelrios** [comment]
> > I would change the name for the github action: `.github/workflows/focus-ios-ui-tests-old.yml` looks like it is something old not running.. I would rather add `focus-ios-ui-tests-previous-os-15-16` or something along these lines.. thanks
> 
> I would say `focus-ios-ui-tests-previous-ios.yml` so that we don't need to update the filename when iOS 15 is no longer supported.

that is a good point, maybe `focus-ios-ui-tests-on-previous-os-versions.yml` then

**Comment by clarmso** [comment]
Note: There will be another PR to add the `schedule` for the trigger. I would like to test the Slack notification using `workflow_dispatch` first before scheduling this job to run regularly.

**Pull request merged**